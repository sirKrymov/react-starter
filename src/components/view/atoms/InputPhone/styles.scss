/* stylelint-disable selector-max-compound-selectors, max-nesting-depth */
@import '../../../../styles/configs';

.input-phone {
  min-height: inherit;

  .phone-input-container {
    .form-control + div::before {
      display: none;
    }
  }

  .react-tel-input {
    min-height: inherit;

    .country-list {
      animation: mount $anim-time;
      margin-top: 2px;
      width: 100%;
      right: 0;
      left: 0;
      top: 112%;
    }

    .form-control {
      background-color: var(--bgPrimary);
      border-radius: $border-radius;
      padding-left: 64px;
      min-height: inherit;
      padding: 0 16px 0 58px;
      border: 1px solid var(--borderPrimary);
      height: initial;
      width: 100%;

      &::placeholder {
        color: var(--textHint);
      }

      &:hover {
        border-color: var(--borderPrimary);
      }

      &:focus {
        border-color: var(--thirdy);
        box-shadow: none;
      }
    }

    .flag-dropdown {
      position: static;
    }

    .form-control + div::before {
      display: none;
    }

    .selected-flag .arrow {
      background-size: contain;
      // background: url('/images/icons/png/select-arrow@2x.png') no-repeat;
      border: 0;
      height: 6px;
      width: 9px;
      left: 32px;
    }

    .selected-flag .arrow.up {
      border: 0;
    }

    .selected-flag {
      position: static;
    }
  }
}
