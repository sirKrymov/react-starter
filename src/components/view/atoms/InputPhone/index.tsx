// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactElement } from 'react';

import ReactPhoneInput from 'react-phone-input-2';
import cn from 'classnames';

import { UiSize } from 'components/view/atoms/UiSize';

import './styles.scss';

interface IProps {
  onChange(value: string): void;
  className?: string;
  disabled?: boolean;
  height?: 'xs' | 'sm' | 'md' | 'lg' | 'full';
  width?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'full' | 'content';
  value: string;
  id: string;
}

export function InputPhone({
  className,
  disabled,
  onChange,
  height = 'md',
  width,
  value,
  id
}: IProps): ReactElement {
  const inputPhoneClass = cn(
    {
      'input-phone': true
    },
    className
  );

  return (
    <UiSize width={width} height={height}>
      <div className={inputPhoneClass} id={id}>
        <ReactPhoneInput
          // countryCodeEditable={false}
          inputProps={{ autoComplete: 'off' }}
          onChange={onChange}
          disabled={disabled}
          // country="in"
          value={value}
        />
      </div>
    </UiSize>
  );
}
