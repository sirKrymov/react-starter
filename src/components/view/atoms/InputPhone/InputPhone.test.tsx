import React from 'react';

import renderer, { ReactTestRenderer } from 'react-test-renderer';

import { InputPhone } from '../InputPhone';

describe('<InputPhone> compound component', () => {
  const handleChange = jest.fn();

  let component: ReactTestRenderer;

  const createComponent = () => {
    component = renderer.create(
      <InputPhone onChange={handleChange} value="" id="check-input-phone" />
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
