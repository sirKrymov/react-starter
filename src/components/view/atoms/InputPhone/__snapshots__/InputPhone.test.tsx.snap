// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPhone> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="ui-size ui-size--height-md ui-size--width-full"
>
  <div
    className="input-phone"
    id="check-input-phone"
  >
    <div
      className=" react-tel-input"
      onKeyDown={[Function]}
      style={Object {}}
    >
      <input
        autoComplete="off"
        className=" form-control"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onCopy={[Function]}
        onDoubleClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="1 (702) 123-4567"
        style={Object {}}
        type="tel"
        value=""
      />
      <div
        className=" flag-dropdown"
        role="button"
        style={Object {}}
        tabIndex="0"
      >
        <div
          className="selected-flag"
          onClick={[Function]}
          title=""
        >
          <div
            className="flag 0"
          >
            <div
              className="arrow"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
